python/lsst/daf/persistence/posixStorage.py

```diff
@@ -709,8 +711,11 @@ def writeFitsCatalogStorage(butlerLocation, obj):
     locations = butlerLocation.getLocations()
     with SafeFilename(os.path.join(butlerLocation.getStorage().root, locations[0])) as locationString:
         logLoc = LogicalLocation(locationString, additionalData)
-        flags = additionalData.getInt("flags", 0)
-        obj.writeFits(logLoc.locString(), flags=flags)
+        if additionalData.exists("flags"):
+            kwds = dict(flags=additionalData.getInt("flags"))
+        else:
+            kwds = {}
```

Comment:
Do we want another `if` here for mode="a"? I don't know if there's any way to include "appending" (instead of overwriting) in a butler call.

Reply:
so the same thing here: IMO the fix in this ticket should only be a correct factoring of Jim's commit and if you want to improve on the changes he made I'd like to address that in a separate ticket.

Reply:
I don't think we never want the butler to modify a dataset in-place.  If we did, we'd need to first figure  out how that interacts with provenance.